### PR TITLE
dev-util/premake: bump version, remove hardcoded -L/usr/lib64

### PR DESCRIPTION
**Pull request by littlewu2508**
Currrently premake hardcoded -L/usr/lib64 as LDFLAG on linux 64 bit
platform. This should be prefixified.

Signed-off-by: YiyangWu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

File: dev-util/premake/premake-5.0.0_alpha16.ebuild [hunk 1]
```diff
@@ -20,6 +21,15 @@ PATCHES=(
 	"${FILESDIR}"/${PN}-5.0.0_alpha16-respect-MAKEOPTS-jobs.patch
 )
 
+src_prepare() {
+	default
+	if use prefix; then
+		grep -rl '\-L/usr' ${S} | while IFS= read srcfile; do
```
Comment (heroxbd) [inline dev-util/premake/premake-5.0.0_alpha16.ebuild:27]: How many affected files are there?  If it is less than 5.  Writing them out explicitly is more robust.

**Review by heroxbd — CHANGES_REQUESTED** [review]

File: dev-util/premake/premake-5.0.0_alpha16.ebuild (hunk 1 above)
Comment (thesamesam) [reply dev-util/premake/premake-5.0.0_alpha16.ebuild:27]: Also, really, I'd prefer that we just patch this out, as it's wrong in other contexts anyway. It's usually not needed to specify it at all if the compiler knows the search paths.

Please only use seds like this as a last resort, as they're hard to test / validate.

**Comment by littlewu2508** [comment]
On Fri, Jul 08, 2022 at 11:52:35PM -0700, Sam James wrote:
> @thesamesam commented on this pull request.
> 
> 
> 
> > @@ -20,6 +21,15 @@ PATCHES=(
>  	"${FILESDIR}"/${PN}-5.0.0_alpha16-respect-MAKEOPTS-jobs.patch
>  )
>  
> +src_prepare() {
> +	default
> +	if use prefix; then
> +		grep -rl '\-L/usr' ${S} | while IFS= read srcfile; do
> 
> Also, really, I'd prefer that we just patch this out, as it's wrong in other contexts anyway. It's usually not needed to specify it at all if the compiler knows the search paths.

Yes, compiler knows the default search paths. Premake detects whether
the system is x86 or x86_64 and set the corresponding -L/usr/lib{,64}

> 
> Please only use seds like this as a last resort, as they're hard to test / validate.

It's just patch out would be harder to maintain. Just inserting an
EPREFIX does minimum changes, I suppose.

**Comment by thesamesam** [comment]
Patching out is the right thing to do though. The sed may silently stop working in future, or have unintended consequences. Plus, if you're only doing it for prefix, nobody is going to notice if it breaks.

Having a patch which is useful _anyway_ (no developer will mind rebasing this, just include an explanation at the top) for non-prefix cases isn't a problem.

**Comment by littlewu2508** [comment]
> Patching out is the right thing to do though. The sed may silently stop working in future, or have unintended consequences. Plus, if you're only doing it for prefix, nobody is going to notice if it breaks.
> 
> Having a patch which is useful _anyway_ (no developer will mind rebasing this, just include an explanation at the top) for non-prefix cases isn't a problem.

Maybe I misunderstood "patching out" previously. I thought you proposed
to remove the "-L/usr/lib64" via patch because it's not needed. That
also affects non-prefix cases. Or do you mean to use conditional patch
(but conditional patch is also a deprecated way)?

So maybe using patch and `eprefixify` to replace the sed command would
be more robust, and has no modification on non-prefix system.

**Comment by thesamesam** [comment]
Hardcoding -L/usr/lib64 is wrong for:
1. ROOT installs
2. Odd libdirs (think about e.g. riscv)
3. Multilib (not that it matters for this ebuild)
4. Prefix
5. ...

It should be removed unconditionally.

**Comment by heroxbd** [comment]
Sam James ***@***.***> writes:

> Hardcoding -L/usr/lib64 is wrong for:
>
> 1 ROOT installs
> 2 Odd libdirs (think about e.g. riscv)
> 3 Multilib (not that it matters for this ebuild)
> 4 Prefix
> 5 ...
>
> It should be removed unconditionally.

I agree with Sam.

**Comment by littlewu2508** [comment]
I have reworked this PR by patching out hardcoded `-L/usr/lib` and `-L/usr/lib64`

**Comment by thesamesam** [comment]
Sorry, one more thing: please cleanup old versions in a separate commit.

**Comment by littlewu2508** [comment]
> Sorry, one more thing: please cleanup old versions in a separate commit.

Shall we keep the latest stable version? If so, is this patch supposed to be applied to the stable ebuild as well?

**Comment by thesamesam** [comment]
1. Yes.
2. I'd say no.

**Pull request closed without merging**